File: Examples/WindowsFormsProject/Models/DataEdge.cs
using GraphX;
using GraphX.PCL.Common.Models;

namespace WindowsFormsProject
{
    /* DataEdge is the data class for the edges. It contains all custom edge data specified by the user.
     * This class also must be derived from EdgeBase class that provides properties and methods mandatory for
     * correct GraphX operations.
     * Some of the useful EdgeBase members are:
     *  - ID property that stores unique positive identfication number. Property must be filled by user.
     *  - IsSelfLoop boolean property that indicates if this edge is self looped (eg have identical Target and Source vertices) 
     *  - RoutingPoints collection of points used to create edge routing path. If Null then straight line will be used to draw edge.
     *      In most cases it is handled automatically by GraphX.
     *  - Source property that holds edge source vertex.
     *  - Target property that holds edge target vertex.
     *  - Weight property that holds optional edge weight value that can be used in some layout algorithms.
     */

    public class DataEdge : EdgeBase<DataVertex>
    {
        /// <summary>
        /// Default constructor. We need to set at least Source and Target properties of the edge.
        /// </summary>
        /// <param name="source">Source vertex data</param>
        /// <param name="target">Target vertex data</param>
        /// <param name="weight">Optional edge weight</param>
        public DataEdge(DataVertex source, DataVertex target, double weight = 1)
			: base(source, target, weight)
		{
		}
        /// <summary>
        /// Default parameterless constructor (for serialization compatibility)
        /// </summary>
        public DataEdge()
            : base(null, null, 1)
        {
        }

        /// <summary>
        /// Custom string property for example
        /// </summary>
        public string Text { get; set; }

        #region GET members
        public override string ToString()
        {
            return Text;
        }
        #endregion
    }
}

File: Examples/WindowsDesktop_VB.NET_WinForms_Example/Models/DataEdge.vb

Imports GraphX.PCL.Common.Models

Namespace Models

    Public Class DataEdge
        Inherits EdgeBase(Of DataVertex)
        ''' <summary>
        ''' Default constructor. We need to set at least Source and Target properties of the edge.
        ''' </summary>
        ''' <param name="source">Source vertex data</param>
        ''' <param name="target">Target vertex data</param>
        ''' <param name="weight">Optional edge weight</param>
        Public Sub New(source As DataVertex, target As DataVertex, Optional weight As Double = 1)
            MyBase.New(source, target, weight)
        End Sub
        ''' <summary>
        ''' Default parameterless constructor (for serialization compatibility)
        ''' </summary>
        Public Sub New()
            MyBase.New(Nothing, Nothing, 1)
        End Sub

        ''' <summary>
        ''' Custom string property for example
        ''' </summary>
        Public Property Text As String

#Region "GET members"
        Public Overloads Function ToString() As String
            Return Text
        End Function


#End Region
    End Class
End Namespace